backend/src/schema/resolvers/comments.spec.js

```diff
@@ -53,335 +39,243 @@ afterEach(async () => {
-    })
+      it('assigns the authenticated user as author', async () => {
+        await expect(mutate({ mutation: createCommentMutation, variables })).resolves.toMatchObject(
+          {
+            data: { CreateComment: { author: { name: 'Author' } } },
+          },
+        )
+      })
 
-    it('throws an error if postId is sent as an empty string', async () => {
-      createCommentVariablesSansPostId = {
-        postId: '',
-        content: 'this comment should not be created',
-      }
+      describe('comment content is empty', () => {
+        beforeEach(() => {
+          variables = { ...variables, content: '<p></p>' }
+        })
 
-      await expect(
-        client.request(createCommentMutation, createCommentVariablesSansPostId),
-      ).rejects.toThrow('Comment cannot be created without a post!')
-    })
+        it('throw UserInput error', async () => {
+          const { data, errors } = await mutate({ mutation: createCommentMutation, variables })
+          expect(data).toEqual({ CreateComment: null })
+          expect(errors[0]).toHaveProperty('message', 'Comment must be at least 1 character long!')
+        })
+      })
 
-    it('throws an error if postId is sent as an string of empty characters', async () => {
-      createCommentVariablesSansPostId = {
-        postId: '   ',
-        content: 'this comment should not be created',
-      }
+      describe('comment content contains only whitespaces', () => {
+        beforeEach(() => {
+          variables = { ...variables, content: '   <p>   </p>   ' }
+        })
 
-      await expect(
-        client.request(createCommentMutation, createCommentVariablesSansPostId),
-      ).rejects.toThrow('Comment cannot be created without a post!')
-    })
+        it('throw UserInput error', async () => {
+          const { data, errors } = await mutate({ mutation: createCommentMutation, variables })
+          expect(data).toEqual({ CreateComment: null })
+          expect(errors[0]).toHaveProperty('message', 'Comment must be at least 1 character long!')
+        })
+      })
 
-    it('throws an error if the post does not exist in the database', async () => {
-      createCommentVariablesWithNonExistentPost = {
-        postId: 'p2',
-        content: "comment should not be created cause the post doesn't exist",
-      }
+      describe('invalid post id', () => {
+        beforeEach(() => {
+          variables = { ...variables, postId: 'does-not-exist' }
+        })
 
-      await expect(
-        client.request(createCommentMutation, createCommentVariablesWithNonExistentPost),
-      ).rejects.toThrow('Comment cannot be created without a post!')
+        it('throw UserInput error', async () => {
+          const { data, errors } = await mutate({ mutation: createCommentMutation, variables })
+          expect(data).toEqual({ CreateComment: null })
+          expect(errors[0]).toHaveProperty('message', 'Comment cannot be created without a post!')
+        })
+      })
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-09-03T09:35:23Z" title="Tuesday, September 3rd 2019, 11:35:23 am +02:00">Sep 3, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -53,335 +39,243 @@ afterEach(async () => {
   await factory.cleanDatabase()
 })
 
+const createCommentMutation = gql`
+  mutation($id: ID, $postId: ID!, $content: String!) {
+    CreateComment(id: $id, postId: $postId, content: $content) {
+      id
+      content
+      author {
+        name
+      }
+    }
+  }
+`
+const setupPostAndComment = async () => {
+  commentAuthor = await factory.create('User')
+  await factory.create('Post', {
+    id: 'p1',
+    content: 'Post to be commented',
+    categoryIds: ['cat9'],
+  })
+  await factory.create('Comment', {
+    id: 'c456',
+    postId: 'p1',
+    author: commentAuthor,
+    content: 'Comment to be deleted',
+  })
+  variables = {
+    ...variables,
+    id: 'c456',
+    content: 'The comment is updated',
+  }
+}
+
 describe('CreateComment', () => {
   describe('unauthenticated', () => {
     it('throws authorization error', async () => {
-      createCommentVariables = {
+      variables = {
+        ...variables,
         postId: 'p1',
         content: "I'm not authorised to comment",
       }
-      client = new GraphQLClient(host)
-      await expect(client.request(createCommentMutation, createCommentVariables)).rejects.toThrow(
-        'Not Authorised',
-      )
+      const { errors } = await mutate({ mutation: createCommentMutation, variables })
+      expect(errors[0]).toHaveProperty('message', 'Not Authorised!')
     })
   })
 
   describe('authenticated', () => {
     beforeEach(async () => {
-      headers = await login(userParams)
-      client = new GraphQLClient(host, {
-        headers,
-      })
-      createCommentVariables = {
-        postId: 'p1',
-        content: "I'm authorised to comment",
-      }
-      await client.request(createPostMutation, createPostVariables)
+      const user = await neode.create('User', { name: 'Author' })
+      authenticatedUser = await user.toJson()
     })
 
-    it('creates a comment', async () => {
-      const expected = {
-        CreateComment: {
+    describe('given a post', () => {
+      beforeEach(async () => {
+        await factory.create('Post', { categoryIds: ['cat9'], id: 'p1' })
+        variables = {
+          ...variables,
+          postId: 'p1',
           content: "I'm authorised to comment",
-        },
-      }
-
-      await expect(
-        client.request(createCommentMutation, createCommentVariables),
-      ).resolves.toMatchObject(expected)
-    })
-
-    it('assigns the authenticated user as author', async () => {
-      await client.request(createCommentMutation, createCommentVariables)
-
-      const { User } = await client.request(gql`
-        {
-          User(name: "TestUser") {
-            comments {
-              content
-            }
-          }
         }
-      `)
-
-      expect(User).toEqual([
-        {
-          comments: [
-            {
-              content: "I'm authorised to comment",
-            },
-          ],
-        },
-      ])
-    })
-
-    it('throw an error if an empty string is sent from the editor as content', async () => {
-      createCommentVariables = {
-        postId: 'p1',
-        content: '<p></p>',
-      }
-
-      await expect(client.request(createCommentMutation, createCommentVariables)).rejects.toThrow(
-        'Comment must be at least 1 character long!',
-      )
-    })
-
-    it('throws an error if a comment sent from the editor does not contain a single character', async () => {
-      createCommentVariables = {
-        postId: 'p1',
-        content: '<p> </p>',
-      }
-
-      await expect(client.request(createCommentMutation, createCommentVariables)).rejects.toThrow(
-        'Comment must be at least 1 character long!',
-      )
-    })
-
-    it('throws an error if postId is sent as an empty string', async () => {
-      createCommentVariables = {
-        postId: 'p1',
-        content: '',
-      }
-
-      await expect(client.request(createCommentMutation, createCommentVariables)).rejects.toThrow(
-        'Comment must be at least 1 character long!',
-      )
-    })
+      })
 
-    it('throws an error if content is sent as an string of empty characters', async () => {
-      createCommentVariables = {
-        postId: 'p1',
-        content: '    ',
-      }
+      it('creates a comment', async () => {
+        await expect(mutate({ mutation: createCommentMutation, variables })).resolves.toMatchObject(
+          {
+            data: { CreateComment: { content: "I'm authorised to comment" } },
+          },
+        )
+      })
 
-      await expect(client.request(createCommentMutation, createCommentVariables)).rejects.toThrow(
-        'Comment must be at least 1 character long!',
-      )
-    })
+      it('assigns the authenticated user as author', async () => {
+        await expect(mutate({ mutation: createCommentMutation, variables })).resolves.toMatchObject(
+          {
+            data: { CreateComment: { author: { name: 'Author' } } },
+          },
+        )
+      })
 
-    it('throws an error if postId is sent as an empty string', async () => {
-      createCommentVariablesSansPostId = {
-        postId: '',
-        content: 'this comment should not be created',
-      }
+      describe('comment content is empty', () => {
+        beforeEach(() => {
+          variables = { ...variables, content: '<p></p>' }
+        })
 
-      await expect(
-        client.request(createCommentMutation, createCommentVariablesSansPostId),
-      ).rejects.toThrow('Comment cannot be created without a post!')
-    })
+        it('throw UserInput error', async () => {
+          const { data, errors } = await mutate({ mutation: createCommentMutation, variables })
+          expect(data).toEqual({ CreateComment: null })
+          expect(errors[0]).toHaveProperty('message', 'Comment must be at least 1 character long!')
+        })
+      })
 
-    it('throws an error if postId is sent as an string of empty characters', async () => {
-      createCommentVariablesSansPostId = {
-        postId: '   ',
-        content: 'this comment should not be created',
-      }
+      describe('comment content contains only whitespaces', () => {
+        beforeEach(() => {
+          variables = { ...variables, content: '   <p>   </p>   ' }
+        })
 
-      await expect(
-        client.request(createCommentMutation, createCommentVariablesSansPostId),
-      ).rejects.toThrow('Comment cannot be created without a post!')
-    })
+        it('throw UserInput error', async () => {
+          const { data, errors } = await mutate({ mutation: createCommentMutation, variables })
+          expect(data).toEqual({ CreateComment: null })
+          expect(errors[0]).toHaveProperty('message', 'Comment must be at least 1 character long!')
+        })
+      })
 
-    it('throws an error if the post does not exist in the database', async () => {
-      createCommentVariablesWithNonExistentPost = {
-        postId: 'p2',
-        content: "comment should not be created cause the post doesn't exist",
-      }
+      describe('invalid post id', () => {
+        beforeEach(() => {
+          variables = { ...variables, postId: 'does-not-exist' }
+        })
 
-      await expect(
-        client.request(createCommentMutation, createCommentVariablesWithNonExistentPost),
-      ).rejects.toThrow('Comment cannot be created without a post!')
+        it('throw UserInput error', async () => {
+          const { data, errors } = await mutate({ mutation: createCommentMutation, variables })
+          expect(data).toEqual({ CreateComment: null })
+          expect(errors[0]).toHaveProperty('message', 'Comment cannot be created without a post!')
+        })
+      })
```


I guess you removed tests that contain empty strings without `<p>` tags. Did you feel these use cases are already tested by the :point_up: tests?
My motive for adding them was thinking of someone sending a direct request to the database, but maybe they were overkill?

Reply:
> [<img alt="roschaefer" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-09-03T13:41:08Z" title="Tuesday, September 3rd 2019, 3:41:08 pm +02:00">Sep 3, 2019</time>_

---


Yes, I think one test to check for emptiness is enough. If `  <p>   </p>  ` is not valid, I think it's pretty clear that `` or `<p></p>` won't be valid either.